site/content/how-to/traffic-management/response-headers.md

```diff
@@ -0,0 +1,209 @@
+---
+
+[HTTPRoute](https://gateway-api.sigs.k8s.io/api-types/httproute/) filters can modify the headers during the request-response lifecycle. [HTTP Header Modifiers](https://gateway-api.sigs.k8s.io/guides/http-header-modifier/?h=request#http-header-modifiers) can be used to add, modify or remove headers in incoming requests.
+
+1. [ResponseHeaderModifier](https://gateway-api.sigs.k8s.io/guides/http-header-modifier/?h=request#http-response-header-modifier) to alter headers in response before responding to the downstream.
+
+
+In this guide we will modify the headers for HTTP responses when client requests are made. For an introduction to exposing your application, we recommend that you follow the [basic guide]({{< relref "/how-to/traffic-management/routing-traffic-to-your-app.md" >}}) first.
+
+
+## Prerequisites
+
+- [Install]({{< relref "/installation/" >}}) NGINX Gateway Fabric.
+- [Expose NGINX Gateway Fabric]({{< relref "installation/expose-nginx-gateway-fabric.md" >}}) and save the public IP
+  address and port of NGINX Gateway Fabric into shell variables:
+
+   ```text
+   GW_IP=XXX.YYY.ZZZ.III
+   GW_PORT=<port number>
+   ```
+
+{{< note >}}In a production environment, you should have a DNS record for the external IP address that is exposed, and it should refer to the hostname that the gateway will forward for.{{< /note >}}
+
+
+## Response Header Filter
+
+In this guide, we will configure an HTTPRoute with a `ResponseHeaderModifier` filter to modify the HTTP response headers of an application.
+
+
+### Deploy the Headers application
+
+Begin by deploying the example application `headers`:
+
+   ```shell
+   kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/v1.3.0/examples/http-response-header-filter/headers.yaml
+   ```
+
+Verify if the Pod is running in the `default` Namespace:
+
+   ```shell
+   kubectl -n default get pods
+   ```
+
+   ```text
+   NAME                      READY   STATUS    RESTARTS   AGE
+   headers-6f854c478-k9z2f   1/1     Running   0          32m
+   ```
+
+### Deploy the Gateway API Resources for the Header Application
+
+
+The [gateway](https://gateway-api.sigs.k8s.io/api-types/gateway/) resource is typically deployed by the [cluster operator](https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/#roles-and-personas_1). To deploy the gateway:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: gateway
+spec:
+  gatewayClassName: nginx
+  listeners:
+  - name: http
+    port: 80
+    protocol: HTTP
+EOF
+```
+
+### Configure the basic HTTPRoute
+
+This gateway defines a single listener on port 80. Since no hostname is specified, this listener matches on all hostnames.
+
+The [HTTPRoute](https://gateway-api.sigs.k8s.io/api-types/httproute/) is typically deployed by the [application developer](https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/#roles-and-personas_1). To deploy the `headers` HTTPRoute:
```

Comment:
We've linked to HTTPRoute before, I don't think we need to do it multiple times.